venter/admin.py

```diff
@@ -89,48 +89,42 @@ def send_emails(modeladmin, request, queryset):  # pylint: disable=R0201
 
         Recipient List: A list containing authorities to whom the email is to be sent
 
-        The email messages are composed for all the selected complaints and appended to the mailing list.
-        After this, send_mass_mail is used to send a tuple of the stored messages using django's default email backend
+        The email messages are sent using the send_mail() method, and use the four parameters listed above
         """
-        # List containing all the email messages to be sent, is being converted to tuple and sent using "send_mass_mail"
-        mailing_list = []
+        # Variable used to keep track of the number of emails that have not been sent
+        email_block = 0
 
         for item in queryset:
-            # Check if the complaint has a valid authority set
-            if item.authority_email is None:
+            # Check if the complaint has a valid authority set. If none exists, remove the complaint from the queryset
+            # and stop the emails from being sent to them
+            if item.authorities.all().exists() is False:
+                email_block += 1
+                queryset = queryset.exclude(id=item.id)
```

Comment:
This seems soo non-intuitive. What advantage does this have over simply updating `email_status` for individual items in the loop itself, so calling `continue` would skip it anyway. Also, change the condition to use `not` instead (see [this answer](https://stackoverflow.com/a/37104262))

Reply:
The queryset.exclude is important because otherwise it tries to send a message to "None" as well. That issue disappears if i exclude the item from the queryset altogether

Reply:
You aren't using queryset after this at all, so effectively it is an unused variable, since the following part is never executed due to the `continue`.

Reply:
That's fixed. I used the exclude method directly while defining the receivers list and it worked.